### ISSUE TITLE
Named formatting options in filename argument is broken when it contains "/"

### ISSUE THREAD

**Issue by Ykid** [PyTorchLightning/pytorch-lightning#5301]
## 🐛 Bug / Limitation

I think this might be a limitation instead of a bug.

Named formatting options in `filename` argument is broken when it contains "/". "/" is used because I want to group different metrics. 

From [pytorch documentation](https://pytorch.org/docs/stable/tensorboard.html)
> Lots of information can be logged for one experiment. To avoid cluttering the UI and have better result clustering, we can group plots by naming them hierarchically. For example, “Loss/train” and “Loss/test” will be grouped together, while “Accuracy/train” and “Accuracy/test” will be grouped separately in the TensorBoard interface.

for example, if I want to save my model using `xx/iou` metric, I wil do something like the following

```
trainer_config = {}
checkpoint_callback = ModelCheckpoint(
      monitor='xx/iou',
      dirpath="/path/to/dir",
      filename='my_model_{epoch:02d}_{xx/iou:.2f}',
      save_top_k=5,
      mode='max',
  )
trainer_config["callbacks"] = [checkpoint_callback]
trainer = pl.Trainer(**trainer_config)
```

I would expect models are saved with names like `/path/to/dir/my_model_epoch=123_iou=80.ckpt` but the slash in "xx/iou" causes some trouble. The checkpoint file is actually saved to "/path/to/dir/my_model_epoch=123_xx" directory with the name `iou=80.ckpt`.

[Since "/" is not allowed in filenames in linux](https://stackoverflow.com/questions/9847288/is-it-possible-to-use-in-a-filename), I think maybe `xx/iou` should be updated to `xx_iou` to work around the issue ?

### Environment

```
pytorch-lightning      1.1.2
python 3.6.9
```

**Comment by Borda** [comment]
@Ykid I see your point but I would say this is rather a feature than a bug, so would you interested in sending a PR for 1.2?

**Comment by Ykid** [comment]
@Borda yea but my schedule is quite tight recently. let me see if I have time for it in the following weeks.

**Comment by quancs** [comment]
Is there any update for this feature? I also want the same thing.

**Comment by edenlightning** [comment]
@quancs want to try and submit a PR?

**Comment by YodaEmbedding** [comment]
Duplicate of https://github.com/PyTorchLightning/pytorch-lightning/issues/4012
